From: YOKOTA Hiroshi <qykth-git@users.noreply.github.com>
Subject: [PATCH] Update LuaTeX support

* Remove unexpected floating point number to support Lua 5.3+ based LuaTeX
* Update LuaTeX hack

---
 doc/texinfo.tex |  8 ++++----
 doc/txi-ja.tex  | 28 ----------------------------
 2 files changed, 4 insertions(+), 32 deletions(-)

Index: b/doc/texinfo.tex
===================================================================
--- a/doc/texinfo.tex
+++ b/doc/texinfo.tex
@@ -3,7 +3,7 @@
 % Load plain if necessary, i.e., if running under initex.
 \expandafter\ifx\csname fmtname\endcsname\relax\input plain\fi
 %
-\def\texinfoversion{2019-09-08.12}
+\def\texinfoversion{2019-09-20.22}
 %
 % Copyright 1985, 1986, 1988, 1990-2019 Free Software Foundation, Inc.
 %
@@ -1052,7 +1052,7 @@ where each line of input produces a line
             tex.sprint(
               string.format(string.char(0x5c) .. string.char(0x25) .. '03o' ..
                             string.char(0x5c) .. string.char(0x25) .. '03o',
-                            (c / 256), (c % 256)))
+                            math.floor(c / 256), math.floor(c % 256)))
           else
             c = c - 0x10000
             local c_hi = c / 1024 + 0xd800
@@ -1062,8 +1062,8 @@ where each line of input produces a line
                             string.char(0x5c) .. string.char(0x25) .. '03o' ..
                             string.char(0x5c) .. string.char(0x25) .. '03o' ..
                             string.char(0x5c) .. string.char(0x25) .. '03o',
-                            (c_hi / 256), (c_hi % 256),
-                            (c_lo / 256), (c_lo % 256)))
+                            math.floor(c_hi / 256), math.floor(c_hi % 256),
+                            math.floor(c_lo / 256), math.floor(c_lo % 256)))
           end
         end
       end
Index: b/doc/txi-ja.tex
===================================================================
--- a/doc/txi-ja.tex
+++ b/doc/txi-ja.tex
@@ -191,21 +191,6 @@
         #1%
         \parseargline\empty% Insert the \empty token, see \finishparsearg below.
     }
-    % Re-define texinfo.tex's \comment
-    \def\comment{\begingroup \catcode`\^^M=\active%
-      \ifx\ltjlineendcomment\thisisundefined
-        % Ignore U+FFFFF for LuaTeX-ja <= 20160208.0
-        \catcode"FFFFF=9%
-      \else
-        % Ignore the character \ltjlineendcomment for LuaTeX-ja > 20160208.0
-        \catcode\ltjlineendcomment=9%
-      \fi
-    \catcode`\@=\other \catcode`\{=\other \catcode`\}=\other\commentxxx}%
-    % Re-let \comment related macros
-    \let\setfilename=\comment
-    \let\dircategory=\comment
-    \let\definfoenclose=\comment
-    \let\footnotestyle=\comment
     % Re-define texinfo.tex's \c
     \def\c{\begingroup \catcode`\^^M=\active%
       \ifx\ltjlineendcomment\thisisundefined
@@ -218,7 +203,12 @@
     \catcode`\@=\other \catcode`\{=\other \catcode`\}=\other%
     \cxxx}
     % Re-let \c related macro
-    \let\texinfoc=\c
+    \let\comment=\c
+    % Re-let \comment related macros
+    \let\setfilename=\comment
+    \let\dircategory=\comment
+    \let\definfoenclose=\comment
+    \let\footnotestyle=\comment
   \fi % LuaTeX
 
   %
